lib/influx.js

```diff
@@ -0,0 +1,38 @@
+import zlib from 'node:zlib';
+import { promisify } from 'node:util';
+
+/*
+* Sends compressed request metrics to InfluxDB
+* https://docs.influxdata.com/enterprise_influxdb/v1/guides/write_data/
+*/
+export const reportMetric = async (request, response, env) => {
+  const compress = promisify(zlib.gzip);
+
+  // Define API endpoint and headers
+  const url = `${env.INFLUX_URL}/api/v2/write?&bucket=${env.INFLUX_DATABASE}&precision=ms`;
+
+  return fetch(url, {
+    method: 'POST',
+    headers: {
+      'Authorization': `Token ${env.INFLUX_TOKEN}`,
+      'Content-Encoding': 'gzip',
+      'Content-Type': 'text/plain; charset=utf-8',
+    },
+    body: await compress(createMetricsFromRequest(request, response, env)),
+  })
+}
+
+/*
+* Returns request metrics in InfluxDB line protocol format
+* https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/ 
+*/
+export const createMetricsFromRequest = (request, response = {}, env) => {
+  const url = new URL(request.url);
+  const timestamp = Date.now();
+  const apiKey = request.headers.get('api-key') || url.searchParams.get('api-key') || (request.headers.get('Authorization')?.startsWith('Bearer ') ? request.headers.get('Authorization').substring(7) : 'unknown');
```

Comment:
I find this line too difficult to parse, let's split it please.

E.g.

```suggestion
  const apiKey = request.headers.get('api-key') || 
    url.searchParams.get('api-key') || (
      request.headers.get('Authorization')?.startsWith('Bearer ') 
      ? request.headers.get('Authorization').substring(7) 
      : 'unknown'
    );
```